src/edu/csus/ecs/pc2/ui/EditProblemPane.java

```diff
@@ -690,232 +700,258 @@ public void enableUpdateButton() {
+                    // check for changes in the executable program name
+                    if ((problemSettings.getCustomValidatorProgramName() == null ^ changedProblemSettings.getCustomValidatorProgramName() == null)) {
+                        // XOR=1 -> they are different(one is null, the other is not) -> something changed
+                        changed = true;
+                    } else if (!(problemSettings.getCustomValidatorProgramName() == null && changedProblemSettings.getCustomValidatorProgramName() == null)) {
+                        // they're not BOTH null (and therefore BOTH are NON-null); something MIGHT have changed
+                        if (!(problemSettings.getCustomValidatorProgramName().equals(changedProblemSettings.getCustomValidatorProgramName()))) {
                             changed = true;
-                        } else if (!(problemSettings.getCustomValidatorProgramName() == null && changedProblemSettings.getCustomValidatorProgramName() == null)) {
-                            // they're not BOTH null (and therefore BOTH are NON-null); something MIGHT have changed
-                            if (!(problemSettings.getCustomValidatorProgramName().equals(changedProblemSettings.getCustomValidatorProgramName()))) {
-                                changed = true;
-                            }
-                        }
-
-                        // check for changes in the actual validator program file
-                        String changedProblemValidatorFileName = changedProblemSettings.getCustomValidatorProgramName();
-                        if (changedProblemValidatorFileName != null && changedProblemValidatorFileName.length() > 0) {
-                            // TODO: there could be a problem here; pdf.getValidatorFile() seems to sometimes return null even though the problem has a validator...
-                            if (!fileSameAs(pdf.getOutputValidatorFile(), changedProblemValidatorFileName)) {
-                                changed = true;
-                                fileChanged++;
-                            }
                         }
+                    }
 
-                        // check for changes in the specified interface being used
-                        if (problemSettings.isUseClicsValidatorInterface() != changedProblemSettings.isUseClicsValidatorInterface()
-                                || problemSettings.isUsePC2ValidatorInterface() != changedProblemSettings.isUsePC2ValidatorInterface()) {
+                    // check for changes in the actual validator program file
+                    String changedProblemValidatorFileName = changedProblemSettings.getCustomValidatorProgramName();
+                    if (changedProblemValidatorFileName != null && changedProblemValidatorFileName.length() > 0) {
+                        // TODO: there could be a problem here; pdf.getValidatorFile() seems to sometimes return null even though the problem has a validator...
```

Comment:
for this TODO, is the code updated to avoid this potential bug ?

Reply:
I don't know.  This TODO was already in the code from long ago; I didn't add it and it has nothing to do with Memory Limits.